lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -274,33 +274,35 @@ sub _schedule_iso {
-            # the file at that URL before running the job
-            my $download_list = create_downloads_list($settings);
-            foreach my $url (keys %$download_list) {
-                my $download_parameters = $download_list->{$url};
-
-        # caveat: The extraction parameter is currently not processed per destination.
-        # If multiple destinations for the same download have a different 'do_extract' parameter the first one will win.
-                unless ($tmp_downloads{$url}) {
-                    $tmp_downloads{$url}
-                      = {destination => {$download_parameters->[0] => 1}, do_extract => $download_parameters->[1]};
-                    next;
-                }
-                $tmp_downloads{$url}->{destination}->{$download_parameters->[0]} = 1
-                  unless ($tmp_downloads{$url}->{destination}->{$download_parameters->[0]});
-            }
-
             # create a new job with these parameters and count if successful, do not send job notifies yet
             try {
-                my $job = $jobs_resultset->create_from_settings($settings, $self->id);
+                # Any setting name ending in _URL is special: it tells us to download
+                # the file at that URL before running the job
+                my $download_list = create_downloads_list($settings);
+                my $job           = $jobs_resultset->create_from_settings($settings, $self->id);
                 push @created_jobs, $job;
-
+                my $j_id = $job->id;
                 $job_ids_by_test_machine{_settings_key($settings)} //= [];
-                push @{$job_ids_by_test_machine{_settings_key($settings)}}, $job->id;
+                push @{$job_ids_by_test_machine{_settings_key($settings)}}, $j_id;
 
                 # set prio if defined explicitely (otherwise default prio is used)
                 $job->update({priority => $prio}) if (defined($prio));
+
+                foreach my $url (keys %$download_list) {
+                    my $download_parameters = $download_list->{$url};
+                    # caveat: The extraction parameter is currently not processed per destination.
+                    # If multiple destinations for the same download have a different 'do_extract' parameter the first one will win.
+                    unless ($tmp_downloads{$url}) {
+                        $tmp_downloads{$url} = {
+                            destination    => {$download_parameters->[0] => 1},
+                            do_extract     => $download_parameters->[1],
+                            blocked_job_id => [$j_id]};
+                        next;
+                    }
+                    push @{$tmp_downloads{$url}->{blocked_job_id}}, $j_id;
+                    $tmp_downloads{$url}->{destination}->{$download_parameters->[0]} = 1
+                      unless ($tmp_downloads{$url}->{destination}->{$download_parameters->[0]});
```

Comment:
You could use a variable, e.g. `my $download_info = $tmp_downloads{$url}` instead of repeating `$tmp_downloads{$url}` all the time.